plugins/scaffolder-backend/src/scaffolder/stages/prepare/azure.test.ts

```diff
@@ -100,36 +90,44 @@ describe('AzurePreparer', () => {
         },
       }),
     );
+    const logger = getVoidLogger();
+    await preparer.prepare(mockEntity, { logger });
```

Comment:
I kinda expected a gitFactory to be passed in explicitly. That way you could just give in a mocked one instead of spying, which feels more brittle

Reply:
Same for all the others of course.

That way you don't have to worry about how it's initialized and to pass in a logger etc

Reply:
I probably want to leave it like it is for now, given that the api will change to `Git.forUrl` instead of per provider. So I'm not sure it makes sense to change it to change it back?

Reply:
Sure, leave it for a future tweak.

Reply:
I am a fan of injecting what you need rather than creating it in a constructor though, in general